docs/sources/tempo/configuration/hosted-storage/s3.md

```diff
@@ -1,50 +1,252 @@
 ---
-title: Amazon S3 permissions
+title: Amazon S3 permissions and management
 menuTitle: Amazon S3
-description: Set Amazon S3 permissions for Tempo.
+description: Amazon S3 permissions and configuration options for Tempo.
 aliases:
   - ../../configuration/s3/ # /docs/tempo/<TEMPO_VERSION>/configuration/s3/
 ---
 
-# Amazon S3 permissions
+# Amazon S3 permissions and management
+
+Tempo supports Amazon S3 for both monolithic and distributed modes.
+Some of the supported features include:
+
+- Object layout: custom `bucket` and optional `prefix` to nest objects in a shared bucket.
+- Performance: hedged requests (`hedge_requests_at`, `hedge_requests_up_to`) to reduce long-tail latency.
+- S3-compatible endpoints and addressing styles: configurable `endpoint`, `forcepathstyle`, and `bucket_lookup_type`. Dual-stack DNS via `enable_dual_stack`.
+- Local development: S3-compatible storage such as MinIO.
 
 For configuration options, refer to the storage section on the [configuration](../../#storage) page.
 
 The following authentication methods are supported:
+
 - AWS environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`
 - Static access key and secret credentials specified in `access_key` and `secret_key`
 - MinIO environment variables `MINIO_ACCESS_KEY` and `MINIO_SECRET_KEY`
-- AWS shared credentials [configuration file](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/create-shared-credentials-file.html)
-- MinIO client credentials [configuration file](https://github.com/minio/mc/blob/master/docs/minio-client-configuration-files.md)
-- AWS IAM ([IRSA via WebIdentity](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html),
-- AWS [EC2 instance role](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html))
+- AWS shared credentials [configuration file](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/
+  create-shared-credentials-file.html)
+- MinIO client credentials [configuration file](https://github.com/minio/mc/blob/master/docs/minio-client-configuration-files.
+  md)
+- AWS IAM ([IRSA via WebIdentity](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html)
+- AWS [EC2 instance role](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html)
 - AWS [EKS Pod Identity](https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html)
 
+## Before you begin
+
+Tempo requires the following configuration to authenticate to and access Amazon S3:
+
+- S3 `bucket` name.
+- Credentials provided via one of the supported authentication methods listed above.
+- For S3-compatible providers (including MinIO), set the `endpoint` and any addressing options (`forcepathstyle`, `bucket_lookup_type`) according to the provider's requirements.
+
+For all configuration keys, refer to the storage section on the [configuration](../../#storage) page.
+
+## Sample configuration for Tempo monolithic mode
```

Comment:
This example was also added by GPT. Is this valid or correct?

Reply:
a lot of this is giberish - you can use this as a reference https://github.com/grafana/tempo/blob/main/docs/sources/tempo/configuration/_index.md#storage

Reply:
I removed the section in this file and I'll point people to the tempo-distributed doc for an example of local deployment using MinIO. I'll find another example for GCS deployment and point people to that instead of having the example here.